test/boost/total_order_check.hh

```diff
@@ -22,23 +22,27 @@
 #pragma once
 
 #include <vector>
+#include <compare>
 #include <boost/test/unit_test.hpp>
 #include <seastar/util/variant_utils.hh>
 #include <variant>
 #include "test/lib/log.hh"
+#include "utils/collection-concepts.hh"
 
 template<typename Comparator, typename... T>
 class total_order_check {
     using element = std::variant<T...>;
     std::vector<std::vector<element>> _data;
     Comparator _cmp;
 private:
-    static int sgn(int x) {
+    static int sgn(std::strong_ordering x) {
         return (x > 0) - (x < 0);
     }
-
+    static int sgn(int x) {
+        return sgn(x <=> 0);
+    }
     // All in left are expect to compare with all in right with expected_order result
-    void check_order(const std::vector<element>& left, const std::vector<element>& right, int order) {
+    void check_order(const std::vector<element>& left, const std::vector<element>& right, int_or_strong_ordering auto order) {
```

Comment:
Maybe this is a stupid question (I'm not fluent in concepts yet...), is this "auto" necessary here? Doesn't "int_or_strong_ordering order" work without the "auto"?

Reply:
`int_or_strong_ordering` is a concept, which works on templates. This function is now actually a template (because of the `auto`) and the aforementioned concept restricts the accepted types.

Reply:
A more verbose way of achieving the same would be:
```c++
template <typename Order>
requires int_or_strong_ordering<Order>
void check_order(const std::vector<element>& left, const std::vector<element>& right, Order order) {
```